x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx

```diff
@@ -510,6 +534,12 @@ export const EditRolePage: FunctionComponent<Props> = ({
     }
   };
 
+  const onDescriptionChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setRole({
+      ...role,
+      description: e.target.value,
```

Comment:
Nit: set description to `undefined` if trimmed value is empty
```suggestion
      description: e.target.value.trim().length ? e.target.value : undefined,
```